brainscore_vision/model_helpers/brain_transformation/behavior.py

```diff
@@ -215,3 +219,92 @@ def labels_to_indices(self, labels):
                 indices.append(label2index[label])
             index2label = OrderedDict((index, label) for label, index in label2index.items())
             return indices, index2label
+
+
+class OddOneOut(BrainModel):
+    def __init__(self, identifier: str, activations_model, layer: Union[str, List[str]]):
+        """
+        :param identifier: a string to identify the model
+        :param activations_model: the model from which to retrieve representations for stimuli
+        :param layer: the single behavioral readout layer or a list of layers to read out of.
+        """
+        self._identifier = identifier
+        self.activations_model = activations_model
+        self.readout = make_list(layer)
+        self.current_task = BrainModel.Task.odd_one_out
+        self.similarity_measure = 'dot'
+
+    @property
+    def identifier(self):
+        return self._identifier
+
+    def start_task(self, task: BrainModel.Task):
+        assert task == BrainModel.Task.odd_one_out
+        self.current_task = task
+
+    def look_at(self, triplets, number_of_trials=1):
+        # Compute unique features and image_pathst
+        stimuli = triplets.drop_duplicates(subset=['stimulus_id'])
+        stimuli = stimuli.sort_values(by='stimulus_id')
+
+        image_paths = set(stimuli.stimulus_paths)
+        def _sort_by_number(path):
+            return int(path.stem)
+
+        sorted_paths = sorted(image_paths, key=_sort_by_number)
+        stimuli.stimulus_paths = sorted_paths
```

Comment:
Hi Martin, i am running into some problems with the assertion in ``attach_stimulus_set_meta``.  I started sorting everything etc. but the assertion is still failing.

``stimuli``looks like this before calling ``features = self.activations_model(stimuli, layers=self.readout)``

```
(Pdb) len(stimuli)
1854
(Pdb) stimuli
      stimulus_id          top_down_1     rank  ... WordNet_ID        Wordnet_ID3 word_freq_online
0               0              animal  51507.0  ...  n02082791         aardvark#1               53
1               1                 NaN  34578.0  ...  n02666196           abacus#2              188
2               2  musical instrument  15132.0  ...  n02672831        accordion#1              816
3               3               fruit  16007.0  ...  n12267677            acorn#1             1289
4               4                 NaN      NaN  ...  n02686379  air_conditioner#1              943
...           ...                 ...      ...  ...        ...                ...              ...
1849         1849                 NaN  16792.0  ...  n04612840             yoke#7              692
1850         1850                food  11647.0  ...  n07841345             yolk#1              881
1851         1851              animal  14397.0  ...  n02391049            zebra#1             1066
1852         1852                 NaN  10687.0  ...  n04238321           zipper#1             1478
1853         1853           vegetable  10667.0  ...  n07716358         zucchini#2             2098
```
and
```
(Pdb) stimuli.stimulus_paths[:5]
[PosixPath('/Users/linussommer/.brainio/stimulus_Hebart2023/0.jpg'), PosixPath('/Users/linussommer/.brainio/stimulus_Hebart2023/1.jpg'), PosixPath('/Users/linussommer/.brainio/stimulus_Hebart2023/2.jpg'), PosixPath('/Users/linussommer/.brainio/stimulus_Hebart2023/3.jpg'), PosixPath('/Users/linussommer/.brainio/stimulus_Hebart2023/4.jpg')]
```